src/reducers/vnicProfiles.js

```diff
@@ -0,0 +1,26 @@
+import { fromJS } from 'immutable'
+import { actionReducer } from './utils'
+import { SET_VNIC_PROFILES, ADD_NETWORKS_TO_VNIC_PROFILES } from '../constants'
+
+const initialState = fromJS({})
+
+const vnicProfiles = actionReducer(initialState, {
+  [SET_VNIC_PROFILES] (state, { payload: { vnicProfiles } }) {
+    const result = {}
+    vnicProfiles.forEach(vnicProfile => {
+      result[vnicProfile.id] = vnicProfile
+    })
+    return fromJS(result)
+  },
+  [ADD_NETWORKS_TO_VNIC_PROFILES] (state, { payload: { networks } }) {
+    return state.map((vnicProfile) => {
+      const index = networks.findIndex((network) => network.id === vnicProfile.getIn(['network', 'id']))
+      if (index > -1) {
+        return vnicProfile.set('network', fromJS(networks[index]))
+      }
+      return vnicProfile
```

Comment:
When can flow get here? Isn't it an error state?

I will not block merging of this on clarification, but please respond anyway and optionally push a follow-up patch adding a log message. Thanks.